packages/auth/src/grant/service.ts

```diff
@@ -156,3 +160,23 @@ async function getByContinue(
     return null
   return grant
 }
+
+async function update(
+  deps: ServiceDependencies,
+  grantId: string,
+  updatedAccessRequests: AccessRequest[]
+): Promise<void> {
+  const { accessService, knex } = deps
+  const trx = await Access.startTransaction(knex)
+  try {
+    const existingAccess = await accessService.getByGrant(grantId)
+    await Access.query(trx)
+      .findByIds(existingAccess.map((a) => a.id))
+      .del()
+    await accessService.createAccess(grantId, updatedAccessRequests, trx)
```

Comment:
https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-5.3
```
Fields that aren't included in the request are considered
   unchanged from the original request.
```
I'm not sure that's happening here, but I'm also unsure how a client is supposed to clear optional fields.

Reply:
Here we remove all `Access` objects in `existingAccess` and after this call we "add" the updated Accesses. Since Accesses without any change would be `undefined`, they wouldn't be mapped into `existingAccess` and so the current Access would be retained.

I'll move all of the logic into this call to make it clearer